shell/platform/darwin/ios/framework/Source/FlutterSpellCheckPluginTest.mm

```diff
@@ -161,11 +161,11 @@ - (void)testFindAllSpellCheckSuggestionsForText {
   XCTAssertTrue(capturedResult.count == 2);
   NSDictionary* suggestionsJSON1 = capturedResult.firstObject;
   XCTAssertEqualObjects(suggestionsJSON1[@"startIndex"], @0);
-  XCTAssertEqualObjects(suggestionsJSON1[@"endIndex"], @4);
+  XCTAssertEqualObjects(suggestionsJSON1[@"endIndex"], @5);
```

Comment:
Is this right?
```
This isss misspelled
xxxxx^^^^xx
01234567890
```
Start index is 5, length is 4, end index is 8, not 9.

Reply:
Yeah, the end is exclusive: https://github.com/flutter/flutter/blob/f57fcac717092e9d55d25a63d79505b455719609/packages/flutter/lib/src/services/spell_check.dart#L19

Reply:
https://api.flutter.dev/flutter/dart-ui/TextRange-class.html
> The next index after the characters in this range.

Well that's confusing, as you experienced from the off by one error...  Can you add a comment about that in `toDictionary`?  In Objective-C this is usually expressed with a tuple range of start index + length, that the endIndex is after the range is not intuitive.

Reply:
```
```
Test Case '-[FlutterSpellCheckPluginTest testFindAllSpellCheckSuggestionsForText]' started.
/../../flutter/shell/platform/darwin/ios/framework/Source/FlutterSpellCheckPluginTest.mm:164: error: -[FlutterSpellCheckPluginTest testFindAllSpellCheckSuggestionsForText] : ((suggestionsJSON1[@"endIndex"]) equal to (@4)) failed: ("5") is not equal to ("4")
/../../flutter/shell/platform/darwin/ios/framework/Source/FlutterSpellCheckPluginTest.mm:168: error: -[FlutterSpellCheckPluginTest testFindAllSpellCheckSuggestionsForText] : ((suggestionsJSON2[@"endIndex"]) equal to (@9)) failed: ("10") is not equal to ("9")
Test Case '-[FlutterSpellCheckPluginTest testFindAllSpellCheckSuggestionsForText]' failed (0.061 seconds).
```

Reply:
@cyanglaz I think I did something weird and reverted your test fixes. Can you re-push them?